lib/fizzy/parser_expr.cpp

```diff
@@ -485,12 +481,12 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, FuncIdx f
             std::tie(if_type, pos) = parse_blocktype(pos, end);
 
             control_stack.emplace(Instr::if_, if_type, static_cast<int>(operand_stack.size()),
-                code.instructions.size(), code.immediates.size());
+                code.instructions.size());
 
             // Placeholders for immediate values, filled at the matching end or else instructions.
-            push(code.immediates, uint32_t{0});  // Diff to the else instruction
-            push(code.immediates, uint32_t{0});  // Diff for the immediates.
-            break;
+            code.instructions.push_back(opcode);
+            push(code.instructions, uint32_t{0});  // Diff to the else instruction
+            continue;
```

Comment:
Why are many of these changed to `continue`?

Reply:
The continue skips the `code.instructions.push_back(opcode);` after the switch because you do it here before pushing the immediates.

Reply:
As opcode is always the first to be pushed, would it be better to move `code.instructions.push_back(opcode);` to before the switch?

Reply:
Possible currently, but I'm not sure this will stand long term. In https://github.com/wasmx/fizzy/pull/622 I present a way to skip noop instructions in parsing. Also we can replace `return` with `br`, etc. So I think the control what opcode is pushed if any on the instruction granularity is useful.

Alternatively, we can push the opcode, and the "pop" it or replace it in some cases.